### DISCUSSION TITLE
How exactly does Pro work in self host?

### DISCUSSION THREAD

**Discussion by Shuttleu** [docusealco/docuseal#274] [Q&A]
I make heavy use of this software and thought I'd see if I can help out with development.
I've been looking through the code to see how everything works and noticed there doesn't seem to be any checks for pro. 
Anything which requires pro to function (SSO for example) just has "Upgrade to Pro" in the views with no checks for whether the user already had Pro.

How exactly can you get these features if its not in the code?

**Comment by omohokcoj** [comment]
@Shuttleu Pro is a separate/private module codebase what is used to enhance the open-source app with new features (similar to how Wordpress can be extended with plugins for instance). To get Pro features you need to subscribe for Pro and reinstall a new Docker image that contains the Pro code/features.

**Comment by EduNetEurope** [comment]
@omohokcoj 
Hello Pete, we use DocuSeal on [cloudron](https://www.cloudron.io/). Is it possible for us to upgrade to the pro version?

**Reply by omohokcoj** [reply]
@EduNetEurope unfortunately that's not possible to use Pro with Cloudron - a separate Pro docker image should be installed in order to get Pro features.

**Reply by EduNetEurope** [reply]
Hi @omohokcoj , Thanks for the quick reply! That's what I figured. :-( Are there any plans to change the upgrading system in the future?

**Reply by omohokcoj** [reply]
@EduNetEurope unfortunately we're not planning adding Pro support for Cloudron in the near future

**Reply by EduNetEurope** [reply]
Ok, Thanks!

**Comment by nanoscopic** [comment]
What is the license on the container / code provided when pro is selected?

Is there some agreement that has to be made upgrading to pro?

**Reply by omohokcoj** [reply]
@nanoscopic Pro is distributed under the commercial license.
We made the commercial license very simple, you can find it below:

```
The DocuSeal EE Commercial License (the "Commercial License")
Copyright (c) 2023-2024 DocuSeal LLC

This software and associated documentation files (the "Software") may only be
used if you (and any entity that you represent) have agreed to, and are in
compliance with, the DocuSeal Subscription available at
https://www.docuseal.com/pricing, or other agreements governing the use of the
Software, as mutually agreed by you and, DocuSeal LLC ("Licensor"), and
otherwise have a valid DocuSeal subscription ("Commercial Subscription") for
the correct number of authorized installations ("Hosts"). You are not granted
any other rights beyond what is expressly stated herein. Subject to the
foregoing, it is forbidden to publish, distribute, sublicense, the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER
IN AN ACTION OF CONTRACT, TORT, OR OTHERWISE, ARISING FROM, OUT OF OR IN
CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```

**Reply by nanoscopic** [reply]
That would seem to allow one to look at what it is doing? Is this intentionally allowed?

This is why I was asking actually. It's obviously unacceptable to duplicate what said code is doing by looking at it while implementing an open source alternative, but it would help to understand what sort of things need to be done in order to add the features provided by pro in an open source fashion.
